### ISSUE TITLE
An array is unexpectedly regarded as a valid ESLint severity

### ISSUE THREAD

**Issue by shinnn** [eslint/eslint#5499]
**What version of ESLint are you using?**

v2.3.0

**What configuration and parser (Espree, Babel-ESLint, etc.) are you using?**

Default parser

**What did you do? Please include the actual source code causing the issue.**

This is an extremely rare case, but a user may carelessly fill an ESLint severity value with an array of the string like below:

```js
const CLIEngine = require('eslint').CLIEngine;

new CLIEngine({
  useEslintrc: false,
  rules: {
    'comma-dangle': [['error'], 'never']
  }
});
```

**What did you expect to happen?**

`new CLIEngine(...)` should throw an error because `['error']` is, of course, not a valid severity according to [the doc](http://eslint.org/docs/user-guide/configuring#configuring-rules).

**What actually happened? Please include the actual, raw output from ESLint.**

`new CLIEngine(...)` doesn't throw any errors, because,

1. [`/^(?:off|warn|error)$/i.test(severity)`](https://github.com/eslint/eslint/blob/951117a93b368db8d5abb782875faf478793b820/lib/config/config-validator.js#L86) always stringify non-string `severity` before testing.
2. `String(['error'])` returns `'error'`. Unfortunately, that also matches `/^(?:off|warn|error)$/i`.

We should do stricter type checking for `severity` to fix this unexpected behavior. I can create a pull request for this issue.

**Comment by ilyavolodin** [comment]
Looks like an oversight to me.

**Comment by BYK** [comment]
:+1: for being more strict and throwing an error for this case.

**Comment by nzakas** [comment]
I suspect this code path isn't being run through the validator.

**Comment by shinnn** [comment]
I've just started to implement strict severity checking.

**Comment by shinnn** [comment]
Submitted a PR. https://github.com/eslint/eslint/pull/5527
